ocaml/testsuite/tests/typing-local/local.ml

```diff
@@ -2569,3 +2569,67 @@ Error: This local value escapes its region
   Hint: This is a partial application
         Adding 1 more argument will make the value non-local
 |}]
+
+(* Escaping uncurried functions *)
```

Comment:
Assuming these are the same as upstream. Did not review carefully.

Reply:
These tests are new: I added them in the first commit because I noticed there were no tests where a function was constrained at the local mode.

I don't add the upstream tests in this PR because we haven't yet changed the way that arity works, so there's nothing new to test.

Reply:
Richard and I looked at these tests together and we think they look fine, so I'll resolve this comment.